core/sql/optimizer/OptPhysRelExpr.cpp

```diff
@@ -3910,6 +3910,20 @@ InputPhysicalProperty* NestedJoin::generateIpp(
   InputPhysicalProperty* ipp = NULL;
 
   // ----------------------------------------------------------------
+  // If there is a cardinality constraint on the outer child, we'll
+  // pass that information along to the inner child.
+  // ----------------------------------------------------------------
+  Cardinality minRows;
+  Cardinality maxRows;
+  CardConstraint * outerCardinalityConstraint = NULL;
+
+  if (getGroupAttr()->hasCardConstraint(minRows, maxRows))
```

Comment:
I would have expected that this would examine the join's left child's group attributes, not those of the join itself?